src/Razor/test/Microsoft.NET.Sdk.Razor.Test/IntegrationTests/MSBuildProcessManager.cs

```diff
@@ -142,6 +127,87 @@ void Process_OutputDataReceived(object sender, DataReceivedEventArgs e)
                     output.AppendLine(e.Data);
                 }
             }
+
+            async Task<ProcessResult> GetTimeoutForProcess(Process process, TimeSpan? timeout)
+            {
+                await Task.Delay(timeout.Value);
+
+                // Don't timeout during debug sessions
+                while (Debugger.IsAttached)
+                {
+                    Thread.Sleep(TimeSpan.FromSeconds(1));
+                }
+                if (!process.HasExited)
+                {
+                    var procDumpProcess = await CaptureDump(process);
+                    if (procDumpProcess != null && procDumpProcess.HasExited)
+                    {
+                        Console.WriteLine("ProcDump failed to run.");
+                        procDumpProcess.Kill();
+                    }
+
+                    // This is a timeout.
+                    process.Kill();
+                }
+
+                throw new TimeoutException($"command '${process.StartInfo.FileName} {process.StartInfo.Arguments}' timed out after {timeout}. Output: {output.ToString()}");
+            }
+
+            async Task<Process> CaptureDump(Process process)
+            {
+                var metadataAttributes = Assembly.GetExecutingAssembly()
+                    .GetCustomAttributes<AssemblyMetadataAttribute>();
+
+                var procDumpPath = metadataAttributes
+                    .SingleOrDefault(ama => ama.Key == "ProcDumpPath")?.Value;
+
+                if (string.IsNullOrEmpty(procDumpPath))
+                {
+                    Console.WriteLine("ProcDumpPath not defined.");
+                    return null;
+                }
+                var procDumpExePath = Path.Combine(procDumpPath, "procdump.exe");
+                if (!File.Exists(procDumpExePath))
+                {
+                    Console.WriteLine($"Can't find procdump.exe in '{procDumpPath}'.");
```

Comment:
Should this be an exception?

Reply:
No, I want to avoid failing hard, if it hangs and we don't see a dump it's obvious that something in here went wrong.

Reply:
Even this is a bit noisy when building locally. Should do nothing when the file does not exist.

Reply:
As a general principle you want to log each step along the way to easily reach a conclusion when things stop working for any reason. I'm saying this because we've had different points of view in the past regarding logging and I want to make very clear that I rather log more information than less.

This logging will happen only when there is a hanging child process and only if things are not correctly configured. In that situation I want to log each step along the way so that we can troubleshoot the issue without having to perform code changes and wait for the issue to arise again.